internal/implementations/gateway/gateway.go

```diff
@@ -2,41 +2,65 @@ package implementation
 
 import (
 	"github.com/go-logr/logr"
+	"github.com/nginxinc/nginx-kubernetes-gateway/internal/events"
+	"k8s.io/apimachinery/pkg/types"
 	"sigs.k8s.io/gateway-api/apis/v1alpha2"
 
 	"github.com/nginxinc/nginx-kubernetes-gateway/internal/config"
 	"github.com/nginxinc/nginx-kubernetes-gateway/pkg/sdk"
 )
 
 type gatewayImplementation struct {
-	conf config.Config
+	conf    config.Config
+	eventCh chan<- interface{}
 }
 
-func NewGatewayImplementation(conf config.Config) sdk.GatewayImpl {
+func NewGatewayImplementation(conf config.Config, eventCh chan<- interface{}) sdk.GatewayImpl {
 	return &gatewayImplementation{
-		conf: conf,
+		conf:    conf,
+		eventCh: eventCh,
 	}
 }
 
 func (impl *gatewayImplementation) Logger() logr.Logger {
 	return impl.conf.Logger
 }
 
-func (impl *gatewayImplementation) ControllerName() string {
-	return impl.conf.GatewayCtlrName
-}
-
 func (impl *gatewayImplementation) Upsert(gw *v1alpha2.Gateway) {
-	if gw.Name == impl.ControllerName() {
-		impl.Logger().Info("Found correct Gateway resource",
+	if gw.Namespace != impl.conf.GatewayNsName.Namespace || gw.Name != impl.conf.GatewayNsName.Name {
+		impl.Logger().Info("Gateway was upserted but ignored based on its namespace/name",
```

Comment:
I think we should add the expected ns name to this log so the user has the ability to fix the error.

Reply:
I don't think this always will be an error though. Because let's say the admin deploys multiple NGINX Gateways, each watching its own Gateway resources.

Reply:
True. Once we support configurable gateway names we can change this back.